streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java

```diff
@@ -520,7 +521,7 @@ public synchronized <K, V> GlobalKTable<K, V> globalTable(final String topic,
      */
     public synchronized StreamsBuilder addStateStore(final StoreBuilder<?> builder) {
         Objects.requireNonNull(builder, "builder can't be null");
-        internalStreamsBuilder.addStateStore(builder);
+        internalStreamsBuilder.addStateStore(new StoreBuilderWrapper<>(builder));
```

Comment:
Right off the bat I don't _love_ the name `StoreBuilderWrapper`...but I think we both know this was going to be a practice in painful out of control java naming lol

Reply:
I originally tried `DelegatingStoreFactory` but settled on this. Open to suggestions :)